rs/log_consumer/src/consumer.rs

```diff
@@ -0,0 +1,178 @@
+use std::time::Duration;
+
+use anyhow::Result;
+use log::{error, info};
+use rdkafka::config::ClientConfig;
+use rdkafka::consumer::{BaseConsumer, Consumer};
+use rdkafka::{Message, Offset, TopicPartitionList};
+use rmp_serde::decode;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LogMessage<T> {
+    pub payload: T,
+    pub topic: String,
+}
+
+pub struct LogConsumer {
+    inner: BaseConsumer,
+}
+
+impl LogConsumer {
+    pub fn new(broker: &str) -> Result<Self> {
+        let consumer: BaseConsumer = ClientConfig::new()
+            .set("bootstrap.servers", broker)
+            .set("group.id", "log_consumer")
+            // start from the beginning of the topic if no offset is stored
+            .set("auto.offset.reset", "earliest")
+            // manually commit offsets
+            .set("enable.auto.commit", "false")
+            .create()?;
+
+        Ok(LogConsumer { inner: consumer })
+    }
+
+    pub async fn consume_logs(&self) -> Result<usize> {
+        let mut processed_ops = 0;
+
+        while let Some(Ok(result)) = self.inner.poll(Duration::from_millis(500)) {
+            processed_ops += 1;
```

Comment:
if we can assign multiple topics to `inner`, what kind of fairness guarantee can we expect?

Reply:
[this](https://stackoverflow.com/questions/70824320/kafka-consumer-fairness-when-fetching-records-from-its-assigned-partitions) might be useful. Also, [perplexity](https://www.perplexity.ai/search/in-librdkafka-what-fairness-gu-dZKAoBljSjqboOEuy6ecLw)

Looks like it's greedy - that means big topic can starve small topic. We probably should rethink our model

Reply:
maybe if we set `num.poll.records` to be small, it could be proper round robin